From: SeongJae Park <sj@kernel.org>
Date: Sun, 25 Aug 2024 17:44:35 -0700
Subject: [PATCH] ==== per-ctx regions prioritization histogram ====

replace per-quota region priorities histogram buffer with per-context one

Each DAMOS quota (struct damos_quota) maintains a histogram for total
regions size per its prioritization score.  DAMOS calcultes minimum
prioritization score of regions that are ok to apply the DAMOS action to
while respecting the quota.  The histogram is constructed only for the
calculation of the minimum score in damos_adjust_quota() for each quota
which called by kdamond_fn().

Hence, there is no real reason to have per-quota histogram.  Only
per-kdamond histogram is needed, since parallel kdamonds could have
races otherwise.  The current implementation is only wasting the memory,
and can easily cause unintended stack usage[1].

So, introducing a per-kdamond histogram and replacing the per-quota one
with it would be the right solution for the issue.  However, supporting
multiple DAMON contexts per kdamond is still an ongoing work[2] without
a clear estimated time of arrival.  Meanwhile, per-context histogram
could be an effective and straightforward solution having no blocker.
Let's fix the problem first in the way.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/UwyncHBH | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/UwyncHBH

diff --git a/damon_meta_changes/UwyncHBH b/damon_meta_changes/UwyncHBH
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

